frontend/Exence/src/app/shared/display-theme.service.ts

```diff
@@ -0,0 +1,86 @@
+import { Injectable } from '@angular/core';
+import { BehaviorSubject, map, Observable } from 'rxjs';
+
+export enum DisplayTheme {
+	DARK = 'DARK',
+	LIGHT = 'LIGHT'
+}
+
+interface ThemeData {
+	name: DisplayTheme;
+	cssClass: string;
+}
+
+@Injectable({
+	providedIn: 'root',
+})
+export class DisplayThemeService {
+	private _displayTheme: BehaviorSubject<DisplayTheme>;
+	readonly displayTheme$: Observable<DisplayTheme>;
+	readonly isDark$: Observable<boolean>;
+
+	get currentTheme(): DisplayTheme { return this._displayTheme.value; }
+
+	private readonly themes: ThemeData[] = [
+		{ name: DisplayTheme.LIGHT, cssClass: 'theme-light' },
+		{ name: DisplayTheme.DARK, cssClass: 'theme-dark' }
+	];
+
+	constructor() {
+		let themeData = this.getInitialTheme();
+
+		this.setCssClass(themeData);
+
+		this._displayTheme = new BehaviorSubject(themeData.name);
+		this.displayTheme$ = this._displayTheme.asObservable();
+		this.isDark$ = this.displayTheme$.pipe(map(a => a === DisplayTheme.DARK));
+	}
+
+	private getInitialTheme(): ThemeData {
+		let preference = localStorage.getItem('displayTheme') as (DisplayTheme | undefined | null);
```

Comment:
1. let helyett lehet const
2. castolásnál nem kell az undefined, mert szerintem a localStorage.getItem csak null-t ad ha nincs
3. szerintem a returnnél `??` jobb lenne a `||` helyett, mert itt pont arra vagyunk kiváncsiak hogy null-e az érték
```suggestion
		const preference = localStorage.getItem('displayTheme') as (DisplayTheme | null);
```